File: collect_app/src/main/java/org/odk/collect/android/widgets/interfaces/GeoDataRequester.java
package org.odk.collect.android.widgets.interfaces;

import org.javarosa.form.api.FormEntryPrompt;
import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry;

public interface GeoDataRequester {

    void requestGeoPoint(FormEntryPrompt prompt, String answerText, WaitingForDataRegistry waitingForDataRegistry);

    void requestGeoShape(FormEntryPrompt prompt, String answerText, WaitingForDataRegistry waitingForDataRegistry);

    void requestGeoTrace(FormEntryPrompt prompt, String answerText, WaitingForDataRegistry waitingForDataRegistry);
}

File: collect_app/src/main/java/org/odk/collect/android/widgets/interfaces/GeoDataRequester.kt
package org.odk.collect.android.widgets.interfaces

import org.javarosa.form.api.FormEntryPrompt
import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry

interface GeoDataRequester {
    fun requestGeoPoint(
        prompt: FormEntryPrompt,
        answerText: String?,
        waitingForDataRegistry: WaitingForDataRegistry
    )

    fun requestGeoShape(
        prompt: FormEntryPrompt,
        answerText: String?,
        waitingForDataRegistry: WaitingForDataRegistry
    )

    fun requestGeoTrace(
        prompt: FormEntryPrompt,
        answerText: String?,
        waitingForDataRegistry: WaitingForDataRegistry
    )
}
